pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java

```diff
@@ -114,7 +115,15 @@ public static Expression getLiteralExpression(SqlLiteral node) {
         literal.setDoubleValue(node.bigDecimalValue().doubleValue());
       }
     } else {
-      literal.setStringValue(StringUtils.replace(node.toValue(), "''", "'"));
+      // TODO: Support null literal and other types.
+      switch (node.getTypeName()) {
+        case BOOLEAN:
+          literal.setBoolValue(node.booleanValue());
+          break;
+        default:
+          literal.setStringValue(StringUtils.replace(node.toValue(), "''", "'"));
+          break;
+      }
```

Comment:
IMO this can be added later. since @Jackie-Jiang might have some idea regarding bool <-> string casting with or w/o dictionary

Reply:
I feel this is not related to casting?